misc/toml-parser/src/main/java/io/ballerina/toml/internal/parser/TomlParser.java

```diff
@@ -389,27 +399,42 @@ private STNode parseValue() {
     private STNode parseNumericalNode() {
         STNode sign = parseSign();
         STNode token = parseNumericalToken();
-        SyntaxKind kind;
-        if (token.kind == SyntaxKind.DECIMAL_INT_TOKEN) {
-            kind = DEC_INT;
+        SyntaxKind kind = getNodeKindNumericNode(token.kind);
+        return STNodeFactory.createNumericLiteralNode(kind, sign, token);
+    }
+
+    private SyntaxKind getNodeKindNumericNode(SyntaxKind tokenKind) {
```

Comment:
Shall we use something like getNumericLiteralKind as the name?
Le'ts use switch case here also